### PR TITLE
refactor(validation): use zod schemas for POST /accounts validation

### PR DESCRIPTION
**Pull request by cajubelt**
rather than AJV. 

Also marks AJV validation function as deprecated and urges the use of zod.

Note that this PR drops support for throwing a validation error when an unsupported fiat account type is posted. Bundling that in with validation was nifty, though a bit more opinionated. I'm open to ideas on how to preserve that functionality, but it might just be an inherent tradeoff of importing zod schemas from fiatconnect-types, which cannot know which particular fiat account schemas will be supported by a given provider.

### CODE REVIEW

File: src/routes/accounts.ts [hunk 1]
```diff
@@ -52,20 +45,9 @@ export function accountsRouter({
...
       async (
-        req: express.Request<
-          {},
-          {},
-          PostFiatAccountRequestBody<SupportedFiatAccountSchemas>
-        >,
+        _req: express.Request<{}, {}, PostFiatAccountRequestBody>,
         _res: express.Response,
       ) => {
-        // Validate data in body for exact fiat account schema type. The body middleware
```
Comment (cajubelt) [inline src/routes/accounts.ts:62]: I think the zod schema does this

File: src/schema/quote-request-body.ts [hunk 2]
```diff
@@ -28,6 +28,13 @@ export const quoteRequestBodySchema: JSONSchemaType<QuoteRequestBody> = {
       type: 'string',
       nullable: true,
     },
+    preview: {
```
Comment (cajubelt) [inline src/schema/quote-request-body.ts:31]: needed this to fix build error since fiatconnect-types was incremented

File: src/routes/accounts.ts [hunk 3]
```diff
@@ -21,17 +20,11 @@ export function accountsRouter({
...
-      PostFiatAccountRequestBody<SupportedFiatAccountSchemas>
-    >,
+    req: express.Request<{}, {}, PostFiatAccountRequestBody>,
     _res: express.Response,
     next: express.NextFunction,
   ) => {
-    req.body = validateSchema<
-      PostFiatAccountRequestBody<SupportedFiatAccountSchemas>
-    >(req.body, 'PostFiatAccountRequestBodySchema')
+    validateZodSchema(req.body, postFiatAccountRequestBodySchema)
```
Comment (cajubelt) [inline src/routes/accounts.ts:27]: note that this doesn't limit validation to some subset of accepted schemas. I'm ok with dropping that feature, but open to ideas of how to preserve it.

Comment (satish-ravi) [reply src/routes/accounts.ts:27]: did this exist just as an example for any provider to restrict the schemas they support? Don't think there's any other option than manually constructing this with the required set of schemas https://github.com/fiatconnect/fiatconnect-types/blob/main/src/fiat-account.ts#L147-L171

**Review by satish-ravi — APPROVED** [review]

**Review by jh2oman — APPROVED** [review]
Looks pretty easy!

File: src/routes/accounts.ts (hunk 3 above)
Comment (cajubelt) [reply src/routes/accounts.ts:27]: yeah I think so

Comment (cajubelt) [reply src/routes/accounts.ts:27]: and yeah that's all I could think of, too

**Pull request merged**